odxtools/statechart.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+from .odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkId
+from .state import State
+from .statetransition import StateTransition
+from .utils import create_description_from_et
+
+
+@dataclass
+class StateChart:
+    """
+    Corresponds to STATE-CHART.
+    """
+
+    odx_id: OdxLinkId
+    short_name: str
+    long_name: Optional[str]
+    description: Optional[str]
+    semantic: str
+    state_transitions: List[StateTransition]
+    start_state_snref: str
+    states: List[State]
+
+    @property
+    def start_state(self) -> State:
+        return self._start_state
+
+    @staticmethod
+    def from_et(et_element, doc_frags: List[OdxDocFragment]) -> "StateChart":
+        short_name = et_element.findtext("SHORT-NAME")
+        odx_id = OdxLinkId.from_et(et_element, doc_frags)
+        assert odx_id is not None
```

Comment:
throw an odx exception with a message instead of an asset.

Reply:
It is on my list: I'd like to introduce an `odxassert(condition, "error message")` method which does this. but it is IMO easier to keep using plain old `assert` for now and switch everything in one go...